lib/rules/no-loop-func.js

```diff
@@ -48,9 +57,12 @@ function getContainingLoopNode(node) {
             case "FunctionExpression":
             case "FunctionDeclaration":
 
-                // We don't need to check nested functions.
-                return null;
+                // We need to check nested functions only in case of IIFE.
+                if (isIIFE(parent)) {
+                    break;
+                }
```

Comment:
Since not all IIFEs are skipped, this could produce duplicate errors:

```js
/* eslint no-loop-func: 2 */

for (var i = 0; i < 5; i++) {
    (async () => {
        () => i;
    })();
}
```

```
  4:6  error  Function declared in a loop contains unsafe references to variable(s) 'i'  no-loop-func
  5:9  error  Function declared in a loop contains unsafe references to variable(s) 'i'  no-loop-func
```

In this case, only the error on the upper function (line 4) should be reported, as before this change ([Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWxvb3AtZnVuYzogMiAqL1xuXG5mb3IgKHZhciBpID0gMDsgaSA8IDU7IGkrKykge1xuICAgIChhc3luYyAoKSA9PiB7XG4gICAgICAgICgpID0+IGk7XG4gICAgfSkoKTtcbn0iLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)).

Perhaps we could maintain the list of skipped functions in a set, and check here if the functions is in that set.

Reply:
Good catch! Fixed.